src/System.Memory/src/System/Span.cs

```diff
@@ -233,6 +233,107 @@ internal Span(Pinnable<T> pinnable, IntPtr byteOffset, int length)
         }
 
         /// <summary>
+        /// Clears the contents of this span.
+        /// </summary>
+        public unsafe void Clear()
+        {
+            if (_length == 0) { return; }
```

Comment:
Cache _length in a local?

(Also, the prevailing style for these earily outs is "
   if (expression)
       return;
)